runtime/src/bank.rs

```diff
@@ -492,6 +492,7 @@ impl Bank {
         new.update_clock();
         new.update_fees();
         new.update_recent_blockhashes();
+        new.update_incinerator();
```

Comment:
Sorry, this is a bit nuanced, but this update fn could be desired to be called inside [`bank.freeze()`](https://github.com/solana-labs/solana/pull/9815/files#diff-a7549f152920d85fb44e6a784b8e5e1fR808) because this `fn` isn't strictly related to the initialization of new child bank state rather, it's book-keeping of old state.

I thought anything could be done at parent should be done at `freeze()`. So, the state transition is complete within in every slot like this: incinerator is 0 at slot start, it's incremented in the middle of the slot, finally it's really burnt at the freeze().

Reply:
Along with the code move, it's nice to rename fn like `burn_lamports_in_incinerator()` as the `update_` verb is currently reserved to mean really-simple state rolling for `sysvar`s.

Reply:
Put another way, the question is whether observable decrease of `bank.capitalization` should occur at the slot in which burn tx was executed or at the later slot (arbitrary later due to possible fork). I think the former is more intuitive. :)

Reply:
Great point, done